src/transformers/modeling_utils.py

```diff
@@ -6134,19 +6134,20 @@ def caching_allocator_warmup(model: PreTrainedModel, expanded_device_map: dict,
 
     # This will kick off the caching allocator to avoid having to Malloc afterwards
     for device, byte_count in total_byte_count.items():
-        if device.type == "cuda":
-            index = device.index if device.index is not None else torch.cuda.current_device()
-            device_memory = torch.cuda.mem_get_info(index)[0]
+        if device.type in ("cuda", "xpu"):
+            device_module = getattr(torch, device.type, torch.cuda)
```

Comment:
better to use `torch_accelerator_module` to align w/ other codes like https://github.com/huggingface/transformers/blob/565c035a2eb610de9a208ee598c47c75cd21fc70/src/transformers/integrations/finegrained_fp8.py#L337